### ISSUE TITLE
Image = Garbled text

### ISSUE THREAD

**Issue by rlatief** [mike42/escpos-php#58]
Hello Mike,

What could probably be the reason for this?

* GD is enabled
* Text printing works **great**
* Epson TM-U220D USB
* Tested both on Epson driver and generic text only driver
* Windows 7, UniformServer (PHP 5.4)

Both graphics.php and bit-image.php produce the same result, garbled text on where I suppose the images should be.

Any pointers will be very appreciated :)

---

Btw, thank you very much! After days of research and tests, this is **the best and efficient approach on printing to a receipt printer from PHP**.

I can actually live without images, I still prefer this than being able to do images (by printing PDF files using AcroRd32.exe) along with the headaches on setting up page height, dealing with ugly fonts, etc., but I'm still wondering if the garbled images is just a little something I missed.

Thank you again!

**Comment by mike42** [comment]
Hi, you've found a bug (Good work!).

This is the same deal as #7. In short: There's many ways to print an image in ESC/POS, and the two I've implemented here aren't compatible with your printer.

Because this is an Epson printer, there's a good chance I will get add a compatible image function in due course. I've grabbed a snippet from an old email where I recommended some alternative image printing libraries, in case images creep back into your use case before this is fixed, as they might be worth a shot.

----

A good minimal "bit image to print buffer" example is my fork of png2escpos, written in C :
- https://github.com/mike42/png2escpos

And this project is the most complete for escpos image wrangling:
- https://github.com/petrkutalek/png2pos
